### DISCUSSION TITLE
DragValue editable when inside a disabled ui

### DISCUSSION THREAD

**Discussion by thmxv** [emilk/egui#4340] [Q&A] [answered]
I have an issue where the value of a `DragValue` is editable even when inside a UI that is globally disabled. As shown in the following video.

[Screencast from 2024-04-08 17-43-22.webm](https://github.com/emilk/egui/assets/948798/0181bc92-f7e9-4516-9d27-b6ef24c9c344)

I cannot reproduce this on the widget gallery of egui.rs where trying to change a drag value by dragging moves the window. But I still think the source of the issue is in egui.

Should I report a bug or continue searching for the source of the issue (I tried unsuccessfully already)?

**Comment by thmxv** [comment]
I also have the same issue with `ScrollArea` inside the disabled UI. The widgets that reacts to clicks are not impacted by this issue.

**Comment by YgorSouza** [comment, marked as answer]
Are you on the latest version of egui? #4058 was meant to address that, but it was closed because the big refactoring in #4026 fixed the issue more broadly. That was released on version 0.27.0. If you're still having problems on that version, you should open an issue about it, but if it is not reproducible in the demo app, you'll have to provide a minimal reproduce against the current master branch.

**Comment by thmxv** [comment]
> Are you on the latest version of egui?

Yes and no. My app is not but I am trying to debug/reproduce the bug with less code:

```Rust
fn main() {
    let mut main_enabled = false;
    let mut drag_enabled = false;
    let mut drag_value = 0;
    let mut slider_enabled = false;

    let _ = eframe::run_simple_native(
        "Issue test",
        eframe::NativeOptions::default(),
        move |ctx, _frame| {
            egui::CentralPanel::default().show(ctx, |ui| {
                ui.checkbox(&mut main_enabled, "Enabled");
                ui.separator();
                ui.add_enabled_ui(main_enabled, |ui| {
                    ui.horizontal(|ui| {
                        ui.checkbox(&mut drag_enabled, "");
                        ui.add_enabled(
                            drag_enabled,
                            egui::DragValue::new(&mut drag_value),
                        );
                    });
                    ui.horizontal(|ui| {
                        ui.checkbox(&mut slider_enabled, "");
                        ui.add_enabled(
                            slider_enabled,
                            egui::Slider::new(&mut drag_value, 0..=100),
                        );
                    });
                    egui::ScrollArea::vertical().show(ui, |ui| {
                        for _ in 0..100 {
                            ui.label("some text");
                        }
                    });
                });
            });
        },
    );
}

```

With this code I do not have the issue for the `Slider` not the `DragValue` with version of egui version 0.25 nor 0.27 but I have it with version 0.26 (the one my app uses at the moment). However the `ScrollArea` being scrollable even when disabled is an issue I have with the 3 tested versions of egui.
